src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -157,8 +161,18 @@ private List<Leg> mapAccessLeg(AccessPathLeg<TripSchedule> accessPathLeg) {
       .getLegs();
   }
 
-  private Leg mapTransitLeg(Leg prevTransitLeg, TransitPathLeg<TripSchedule> pathLeg) {
-    TripSchedule tripSchedule = pathLeg.trip();
+  private Leg mapTransitLeg(Leg prevTransitLeg, TransitPathLeg<T> pathLeg) {
+    T tripSchedule = pathLeg.trip();
+
+    // If the next leg is an egress leg without a duration then this transit leg
+    // arrives at the destination.
+    // The cost of the egress should therefore be included in the last transit
+    // leg given that it is the last leg.
+    int lastLegCost = 0;
+    PathLeg<T> nextLeg = pathLeg.nextLeg();
+    if (nextLeg.isEgressLeg() && nextLeg.asEgressLeg().egress().stopReachedOnBoard()) {
```

Comment:
Rather than checking if the last leg has zero duration, would it not make sense to somehow check that the alight penalty has been shifted from the transit leg to the egress one?

Reply:
This check is needed in order to choose whether to shift the cost from one leg to the other - so it is difficult to do this the other way around.

Have I interpreted you correctly?

Reply:
My question is: do we only want to shift the cost back if the egress leg is empt/zero durationy? Could there not be an egress leg with a short walking duration, say at a cost of 100, that still has a -3000 penalty, so we end up with a leg with a cost of -2900?